src/plugin/plugin.ts

```diff
@@ -1231,6 +1233,110 @@ class Purchases {
     );
   }
 
+  /**
+   * iOS only. Use this function to retrieve the eligible win-back offers that a subscriber
+   * is eligible for for a given product.
+   *
+   * @param {PurchasesStoreProduct} product The product the user intends to purchase.
+   * @param {function(PurchasesWinBackOffer[]):void} callback Callback triggered after a successful getEligibleWinBackOffersForProduct call.
+   * It will receive an array of eligible win-back objects for the provided product.
+   * @param {function(PurchasesError):void} errorCallback Callback triggered after an error when retrieving eligible win-back offers.
+   */
+  public static async getEligibleWinBackOffersForProduct(
+    product: PurchasesStoreProduct,
+    callback: (winBackOffers: PurchasesWinBackOffer[]) => void,
+    errorCallback: (error: PurchasesError) => void
+  ) {
+    window.cordova.exec(
+      callback,
+      errorCallback,
+      PLUGIN_NAME,
+      "getEligibleWinBackOffersForProduct",
+      [product.identifier]
+    );
+  }
+
+  /**
+   * iOS only. Use this function to purchase a product with a win-back offer. Fetch eligible win-back offers with getEligibleWinBackOffersForProduct.
+   *
+   * @param {PurchasesStoreProduct} product The product the user intends to purchase.
+   * @param {PurchasesWinBackOffer} winBackOffer The win-back offer the user intends to purchase.
+   * @param {function(PurchasesWinBackOffer[]):void} callback Callback triggered after a successful purchaseProductWithWinBackOffer call.
```

Comment:
Is `PurchasesWinBackOffer[]` correct as callback parameter here? It seems like we're returning `productIdentifier, customerInfo`

Reply:
Great catch! I've updated it to have the correct callback params here, and have double checked the response to make sure it matches.